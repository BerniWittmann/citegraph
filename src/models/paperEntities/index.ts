import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
