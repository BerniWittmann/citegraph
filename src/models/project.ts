export default class Project {
  name: string;
  id?: number;

  constructor ({ name, id }: Project) {
    this.id = id
    this.name = name
  }
}
