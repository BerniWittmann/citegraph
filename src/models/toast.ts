import uuid from 'uuid'

export enum ToastSeverity {
  SUCCESS = 'success',
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error'
}

export enum ToastIcon {
  SUCCESS = 'mdi-check-circle-outline',
  INFO = 'mdi-information-outline',
  WARNING = 'mdi-alert',
  ERROR = 'mdi-alert-circle-outline',
}

export enum ToastDelay {
  SUCCESS = 3000,
  INFO = 4000,
  WARNING = 5000,
  ERROR = 6000,
}

interface ToastData {
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number
}

export default class Toast implements ToastData {
  id: string;
  timestamp: number;
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number

  constructor ({ severity, icon, text, delay }: ToastData) {
    this.id = uuid.v4()
    this.timestamp = Date.now()
    this.severity = severity
    this.icon = icon
    this.text = text
    this.delay = delay
  }
}

export class SuccessToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.SUCCESS,
      icon: ToastIcon.SUCCESS,
      delay: ToastDelay.SUCCESS
    })
  }
}

export class InfoToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.INFO,
      icon: ToastIcon.INFO,
      delay: ToastDelay.INFO
    })
  }
}

export class WarningToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.WARNING,
      icon: ToastIcon.WARNING,
      delay: ToastDelay.WARNING
    })
  }
}

export class ErrorToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.ERROR,
      icon: ToastIcon.ERROR,
      delay: ToastDelay.ERROR
    })
  }
}
