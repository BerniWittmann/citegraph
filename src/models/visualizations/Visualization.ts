import Transformer from '@/transformers/Transformer'

export interface TimePeriod {
  from: number
  to: number
}

export interface VisualizationBaseFields {
  id?: string;
  progress?: number;
  name: string;
  timePeriods?: Array<TimePeriod>
}

export interface VisualizationBaseFieldsDTO {
  id?: string;
  progress?: number;
  name: string;
  key: string;
  // eslint-disable-next-line camelcase
  time_periods?: Array<TimePeriod>
}

export default abstract class Visualization implements VisualizationBaseFields {
  id?: string;
  progress?: number;
  name: string;
  timePeriods: Array<TimePeriod>
  abstract parameters: Object;
  abstract data?: Object;
  abstract readonly key: string;
  static readonly key: string;
  static readonly title: string;
  static readonly imageUrl: string;
  static readonly description: string;
  static readonly longDescription: string;
  static readonly transformer: typeof Transformer;

  protected constructor ({ id, progress, name, timePeriods }: VisualizationBaseFields) {
    this.id = id
    this.progress = progress
    this.name = name
    this.timePeriods = timePeriods || []
  }
}
