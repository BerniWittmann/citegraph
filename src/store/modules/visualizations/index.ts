import { Module } from 'vuex'
import { VisualizationsState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const visualizations: Module<VisualizationsState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
