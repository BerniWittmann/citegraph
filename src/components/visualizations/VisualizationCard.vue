<template>
  <v-card :hover="true"
          class="visualization-card"
          height="100%">
    <div class="d-flex flex-no-wrap justify-space-between">
      <v-avatar
        class="ma-3"
        size="125"
        tile
      >
        <v-img contain :src="imageUrl"/>
      </v-avatar>
      <div class="flex-grow-1 d-flex flex-column align-center">
        <v-card-title>{{ visualization.name }}</v-card-title>
        <v-card-text class="progress-information text-center">
          <span v-if="isLoading">{{ $t('visualizations.card.processing', { percent: progress }) }}</span>
          <span v-else>{{ $t('visualizations.card.ready') }}</span>
        </v-card-text>
        <v-divider style="width: 90%" class="mx-auto"></v-divider>
        <v-card-actions>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.show') }}
          </v-btn>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.edit') }}
          </v-btn>
        </v-card-actions>
      </div>
    </div>
    <v-progress-linear color="#F37F21" background-color="#FBBF35" background-opacity="0.5" v-show="isLoading" class="progress" :value="progress" :max="100"></v-progress-linear>
  </v-card>
</template>
<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import Visualization from '../../models/visualizations/Visualization'
import { visualizationsKeyMap } from '@/models/visualizations'

@Component
export default class VisualizationCard extends Vue {
  @Prop() readonly visualization!: Visualization

  get visualizationClass () {
    return visualizationsKeyMap[this.visualization.key]
  }

  get imageUrl (): string {
    return `${process.env.BASE_URL}${this.visualizationClass.imageUrl}`
  }

  get progress (): number {
    return this.visualization.progress ? this.visualization.progress * 100 : 0
  }

  get isLoading (): boolean {
    return !!this.visualization.progress && (this.progress < 100)
  }
}
</script>

<style scoped lang="scss">
.progress-information {
  color: #F37F21;
}
</style>
