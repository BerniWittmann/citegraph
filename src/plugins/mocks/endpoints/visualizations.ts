import { AxiosRequestConfig } from 'axios'
import MockAdapter from 'axios-mock-adapter'
import db from '../db'
import Project from '@/models/project'
import Visualization from '@/models/visualizations/Visualization'

const projects = db.getCollection('projects')
const visualizations = db.getCollection('visualizations')

function getProjectId (config: AxiosRequestConfig): number {
  const urlParts: Array<string> = config.url!.split('/')
  return parseInt(urlParts[4])
}

function getVisualizationId (config: AxiosRequestConfig): string {
  const urlParts: Array<string> = config.url!.split('/')
  return urlParts[6]
}

export function setupMocks (mock: MockAdapter) {
  mock.onGet(/\/projects\/\d+\/visualizations$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    return [200, visualizations.find()]
  })

  mock.onGet(/\/projects\/\d+\/visualizations\/\d+$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    const visualization: Resultset<Visualization> = visualizations.findOne({ 'id': getVisualizationId(config) })
    if (!visualization) {
      return [404]
    }
    return [200, visualization]
  })
}
