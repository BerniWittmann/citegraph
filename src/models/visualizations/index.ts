import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
