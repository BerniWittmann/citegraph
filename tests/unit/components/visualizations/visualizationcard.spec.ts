import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../setupPlugins'

import VisualizationCard from '@/components/visualizations/VisualizationCard.vue'
import Visualization from '@/models/visualizations/Visualization'
import { visualizations } from '@/models/visualizations'

describe('components/visualizations/VisualizationCard.vue', () => {
  const data = {
    id: '1',
    name: 'My awesome Visualization'
  }
  const getWrapper = (visualization: Visualization) => {
    return shallowMount(VisualizationCard, {
      i18n,
      propsData: {
        visualization
      }
    })
  }

  visualizations.forEach(VisualizationClass => {
    describe('can display a card of type ' + VisualizationClass.key, () => {
      it('renders', () => {
        const wrapper = getWrapper(new VisualizationClass(data))
        expect(wrapper.html()).toMatchSnapshot()
      })

      it('renders a progress bar', () => {
        const wrapper = getWrapper(new VisualizationClass({
          ...data,
          progress: 0.3
        }))
        expect(wrapper.html()).toMatchSnapshot()
      })

      it('does not render a progress bar for 100%', () => {
        const wrapper = getWrapper(new VisualizationClass({
          ...data,
          progress: 1
        }))
        expect(wrapper.html()).toMatchSnapshot()
      })

      it('renders the image', () => {
        const wrapper = getWrapper(new VisualizationClass(data))
        const img = wrapper.find('v-img-stub')
        expect(img.exists()).toBeTruthy()
        expect(img.props('src')).toContain(VisualizationClass.imageUrl)
      })
    })
  })
})
