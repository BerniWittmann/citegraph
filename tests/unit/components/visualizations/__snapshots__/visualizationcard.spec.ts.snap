// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/visualizations/VisualizationCard.vue can display a card of type barchart does not render a progress bar for 100% 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="100" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type barchart renders 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="0" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type barchart renders a progress bar 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/bar_chart.png" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.processing</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="30" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type network does not render a progress bar for 100% 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/network.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="100" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type network renders 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/network.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="0" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type network renders a progress bar 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/network.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.processing</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="30" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type wordcloud does not render a progress bar for 100% 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="100" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type wordcloud renders 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.ready</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="0" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress" style="display: none;"></v-progress-linear-stub>
</v-card-stub>
`;

exports[`components/visualizations/VisualizationCard.vue can display a card of type wordcloud renders a progress bar 1`] = `
<v-card-stub loaderheight="4" tag="div" height="100%" hover="true" class="visualization-card">
  <div class="d-flex flex-wrap justify-space-between">
    <div class="ma-3 flex-shrink-0 flex-grow-1 d-flex justify-center">
      <v-avatar-stub size="100" tile="true">
        <v-img-stub contain="true" options="[object Object]" position="center center" src="undefinedvisualization_images/word_cloud.jpg" transition="fade-transition"></v-img-stub>
      </v-avatar-stub>
    </div>
    <div class="flex-grow-1 d-flex flex-column align-center">
      <v-card-title-stub>My awesome Visualization</v-card-title-stub>
      <v-card-text-stub class="progress-information text-center"><span>visualizations.card.processing</span></v-card-text-stub>
      <v-divider-stub class="mx-auto" style="width: 90%;"></v-divider-stub>
      <v-card-actions-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button">
          visualizations.card.show
        </v-btn-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" to="[object Object]" text="true" type="button">
          visualizations.card.edit
        </v-btn-stub>
      </v-card-actions-stub>
    </div>
  </div>
  <v-progress-linear-stub color="#F37F21" value="30" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="4" max="100" class="progress"></v-progress-linear-stub>
</v-card-stub>
`;
